kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -368,4 +523,1496 @@ def _document(cls):
+    per octave by default, but can be set like `Q = (8, 2)`. Internally,
+    `J_fr` and `Q_fr`, the frequential variants of `J` and `Q`, are defaulted,
+    but can be specified as well.
+
+    {parameters}
+
+    {attributes}
+
+    {terminology}
+    """
+
+    _doc_params = \
+    r"""
+    Parameters
+    ----------
+    J, shape, T, average, oversampling, pad_mode :
+        See `help(kymatio.scattering1d.Scattering1D)`.
+
+        Unlike in time scattering, `T` plays a role even if `average=False`,
+        to compute `phi_t` pairs.
+
+    Q : int / tuple[int]
+        `(Q1, Q2)`, where `Q2=1` if `Q` is int. `Q1` is the number of first-order
+        wavelets per octave, and `Q2` the second-order.
+
+          - `Q1`, together with `J`, determines `N_frs_max` and `N_frs`,
+            or length of inputs to frequential scattering.
+          - `Q2`, together with `J`, determines `N_frs` (via `j2 > j1`
+            criterion), and total number of joint slices.
+          - Greater `Q2` values better capture temporal AM modulations (AMM) of
+            multiple rates. Suited for inputs of multirate or intricate AM.
+            `Q2=2` is in close correspondence with the mamallian auditory cortex:
+            https://asa.scitation.org/doi/full/10.1121/1.1945807
+            2 or 1 should work for most purposes.
+
+    J_fr : int
+        The maximum log-scale of frequential scattering in joint scattering
+        transform, and number of octaves of frequential filters. That is,
+        the maximum (bandpass) scale is given by `2**J_fr`.
+
+        Default is determined at instantiation from longest frequential row
+        in frequential scattering, set to `log2(nextpow2(N_frs_max)) - 2`, i.e.
+        maximum possible minus 2, but no less than 3, and no more than max.
+
+    Q_fr : int
+        Number of wavelets per octave for frequential scattering.
+
+        Greater values better capture quefrential variations of multiple rates
+        - that is, variations and structures along frequency axis of the wavelet
+        transform's 2D time-frequency plane. Suited for inputs of many frequencies
+        or intricate AM-FM variations. 2 or 1 should work for most purposes.
+
+    F : int / str['global'] / None
+        Temporal support of frequential low-pass filter, controlling amount of
+        imposed frequency transposition invariance and maximum frequential
+        subsampling. Defaults to `Q`, i.e. one octave.
+
+          - If `'global'`, sets to maximum possible `F` based on `N_frs_max`.
+          - Used even with `average_fr=False` (see its docs); this is likewise
+            true of `T` for `phi_t * phi_f` and `phi_t * psi_f` pairs.
+
+    implementation : int / None
+        Preset configuration to use. Overrides the following parameters:
+
+            - `average_fr, aligned, out_3D, sampling_filters_fr`
+
+        Defaults to `None`, and any `None` argument above will default to
```

Comment:
this sentence is unclear. Can anyone help?

Reply:
if `implementation=None` and ``average_fr | aligned | out_3D | sampling_filters_fr == None` then `implementation=1` is used.

Reply:
If a user is going beyond implementations 2/3, could we assume they have already figured out the parameters to achieve their implementation? What if I want a `3D` output without `average_fr`? 

Is it necessary to have these options? Or would it be better to document possible options and use cases with examples?

Reply:
> Is it necessary to have these options? Or would it be better to document possible options and use cases with examples?

I don't understand, could you elaborate? We have kwargs for full control, and `implementation` for presets. We need not scrap one for other.

> What if I want a 3D output without average_fr?

`out_3D=True; average_fr=False`? @rastegah

Reply:
an `implementation` argument adds overhead. reading about it is quite confusing. By the time i've figured out what all the presets do, I probably already understand each of the arguments well enough to create my own preset. I don't find it so helpful. 

>     What if I want a 3D output without average_fr?

I say this because its an example of a possible preset that is not available. There are plenty of possible presets. I guess that a given user may only ever opt for two of them. 

Overriding parameters could have consequences due to incorrect interpretation by the user. If I set `implementation=1`, `average_fr=True`, `aligned=True`, `out_3D=True`, what do I get? How can I be sure that I get what I ask for?

Does this _preset_ functionality add confusion? Can you use it before understanding the role of the overridden parameters?

What I am saying is, rather than overriding parameters, documentation would better serve to demonstrate how to use the frontend to achieve an _implementation_ and give examples of _when_ you may want to use a particular implementation.

Reply:
@rastegah well, precisey, it's impossible to do 3D output without average_fr because the frequency indices `n1`  would not be evenly sampled across quefrencies `j_fr`, so we couldn't stack them. Having to explain that in the documentation, and to catch those edge cases in the code, would be difficult, due to the combinatorial explosion of low-level logical settings.

The nice thing about `implementation` is that it sets a nice framework of what's possible vs. what isn't, in the form of a collection of officially supported presets.


> Overriding parameters could have consequences due to incorrect interpretation by the user

I agree that overriding parameters would be a disaster and should not be allowed.


>  If I set implementation=1, average_fr=True, aligned=True, out_3D=True, what do I get? 

Good example. I can't resolve the logic in my head at the glance of an eye, which should be enough to tell us that implementation should not coexist with average_fr and friends. It's one XOR the other 😅 



> What I am saying is, rather than overriding parameters, documentation would better serve to demonstrate how to use the frontend to achieve an implementation and give examples of when you may want to use a particular implementation.

Absolutely. This JTFS needs to be documented via a tutorial and illustrating examples, not just docstrings. @OverLordGoldDragon has already done a fantastic work of illustrating examples, and we have people around to help (Changhong, Han, you), so i'm optimistic about that.

Reply:
An argument docstring is meant to document the argument, not explain the entire algorithm. Of course reading `implementation` alone will leave one clueless.

I agree for a need for an intro, with 

>  examples of when you may want to use a particular implementation

and it could be referred to in `implementation`.

The idea behind current `implementation` is simple: if you use it, you will get parameters specified in it no matter what. If want full control, then set it to `None` and pass your own stuff. Regardless I agree there's awkwardness in design, one solved by @lostanlen's suggestion to move it all into `dict` while preserving `int` option - main downside being that arguments are no longer exposed directly in the function's namespace, and one must read docs.

I favor moving the args into `implementation` per the `dict` + `int` design.

Reply:
@rastegah would you be on board with this? `implementation` is either `int` or `dict`

Reply:
> @rastegah would you be on board with this? `implementation` is either `int` or `dict`

this seems like the best solution in the current state. that way, `implementation` itself must be overriden.

let's make sure that the overriden parameters are not `kwargs`